qiskit/algorithms/fidelities/base_fidelity.py

```diff
@@ -0,0 +1,253 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+Base fidelity interface
+"""
+
+from __future__ import annotations
+from abc import ABC, abstractmethod
+from typing import Sequence, Mapping
+import numpy as np
+
+from qiskit import QuantumCircuit
+from qiskit.circuit import ParameterVector
+
+
+class BaseFidelity(ABC):
+    """
+    An interface to calculate fidelities (state overlaps) for pairs of
+    (parametrized) quantum circuits.
+    """
+
+    def __init__(
+        self,
+    ) -> None:
+        """Initializes the class to evaluate fidelities."""
+
+        self._circuits: Sequence[QuantumCircuit] = []
+        self._parameter_values: Sequence[Sequence[float]] = []
+
+        # use cache for preventing unnecessary circuit compositions
+        self._circuit_cache: Mapping[(int, int), QuantumCircuit] = {}
+
+        self._left_parameters = []
+        self._right_parameters = []
```

Comment:
I see why we store the circuit cache, but all other variables seem to be only temporary bound to one single execution. Couldn't we just pass them around instead of storing them? Otherwise the object always has the "old" state stored although it doesn't need it 🤔

Reply:
Yes, I also didn't like having the "old" state lying around. Removed!